### ISSUE TITLE
slds-cell-shrink not working with slds-table--fixed-layout

### ISSUE THREAD

**Issue by istiti** [salesforce-ux/design-system#378]
I try to use `slds-cell-shrink` in one of my td but it doesn't work I think because if  `slds-table--fixed-layout`

Can I use slds-cell-shrink with fixed-layout ? Because I want first column to be with fixed widht and second shrink to his content
```
<table class="slds-table slds-table--fixed-layout">
<thead>
<th class="slds-truncate widht-100px">dddddddd</th> 
<th class="slds-cell-shrink">thiss need have widht depending in this content</th> 
</thead>
</table>
```

It seems not working because width of my slds-cell-shrink is always 3rem this is your css properties 
```
.tsab .slds-table--fixed-layout .slds-cell-shrink {
    width: 3rem;
}
```
I think this not use width of my content, how can I achieve this ? thanks

PS: if needed I can remove `slds-table--fixed-layout` class but how can I set maximum width on my first column?

**Comment by kaelig** [comment]
Thanks for asking, @istiti - can you reproduce this in a pen on CodePen or using the [Starter Kit](https://github.com/salesforce-ux/design-system-starter-kit) so we can investigate?

**Comment by istiti** [comment]
@kaelig have you a plunker template ? if not can you add to your readme a plunker link easy fork it with always last version (I think this will be good idea) 

ps:wait&see my plunker

**Comment by kaelig** [comment]
@istiti Here's a template for you! http://plnkr.co/edit/QM9J0GvleLPDMb2r9xcb

Hope it helps.

**Comment by istiti** [comment]
@kaelig thanks a lot I think you can add link  to readme page before writing issue 
so this is my behavior : http://plnkr.co/edit/fARdZsxPhceBntFRtDuq?p=preview

**Comment by kaelig** [comment]
Thanks!

I know it doesn't fully answer your use case but would this example be good enough for you? https://www.lightningdesignsystem.com/components/data-tables/#flavor-advanced-resized-column

**Comment by istiti** [comment]
@kaelig this is not what I expect from. 
I need to specify size of width to first column  and second column and last column with slds-shrink to Shrinks cell to width of content

**Comment by stefsullrew** [comment]
When I add the `slds-cell-shrink` class to the `th` (shown truncated in this image), it works:

![image](https://cloud.githubusercontent.com/assets/676213/22219501/d4386ae6-e15f-11e6-9e21-b6d0d792993d.png)

**Comment by stefsullrew** [comment]
OH... I just looked at your example. You removed the `slds-truncate` class which means (since it's a table) that it stretches the cell. Leave the `slds-cell-shrink` on the `th` and put the `slds-truncate` class back on the `td` and it works.

Cheers

**Comment by istiti** [comment]
Yes but i dont think this is behavior of shrink class. I want a column who is taking widht depending of content. I need auto width if my column have 5 or 10 or 100characters I want see all my content of this column how achieve if not with this shrink class?

**Comment by stefsullrew** [comment]
The way a table works is that a cell's content will all render. On a normal table (not fixed layout), the whole cell grows. But if a fixed table, the width will actually be fixed, but the content will render anyway. 

If you put the cell-shrink class on, and you have the truncation there, it will shrink. If you do not have the truncation there, then the content will show. It will either grow wider than your table (like your example) or will protrude into another cell. Here's an example of that:

![image](https://cloud.githubusercontent.com/assets/676213/22222767/f507491a-e16c-11e6-98b4-dd9c38b65c64.png)

In other words, what you would like to do is not possible. Tables have certain limitations. You've hit one of them.

Cheers

**Comment by istiti** [comment]
@stefsullrew Wunderfull explanation ! Thanks to be so clear! 
So the documentation tell : when add cell-shrink class, column will have width depending on its content ! This information is a litle incorrect, isn ´t it?
Because the only thing is that if we don't add truncate class the content will protrude into another td ...

**Comment by stefsullrew** [comment]
Glad that helped... and great point, @istiti — I believe that doc piece was written prior to creating the fixed tables... I'll put it in here as a bug so we can get it fixed up. Cheers.
